Automattic/woocommerce-payments
# Add support for button styling for ECE buttons

Fixes #8951 

## Changes proposed in this Pull Request









- Respect Apple Pay and Google Pay button settings when using ECE buttons.

No automated tests because it's hard to make assertions on iframes and their style.

## Testing instructions




> [!IMPORTANT]
> There's an issue with Storefront that causes the button width to be incorrect. We're still discussing how to address that. In the meantime you can test with Twenty-Twenty Four. See https://github.com/Automattic/woocommerce-payments/issues/8951#issuecomment-2183378659 for details.

> [!IMPORTANT]
> HTTPS is a **hard requirement** for testing the buttons. They will not be rendered unless you have a valid HTTPS URL to test with. Jurassic tube is recommended for permanent custom URLs.

### Setup

1. Make sure you have a valid HTTPS URL you can use for testing the buttons.
2. Register the domain for Apple Pay with the following CLI command:

```bash
curl https://api.stripe.com/v1/payment_method_domains \
     # Don't forget the `:` at the end!
  -u "<secret key, sk_xxx or rk_xxx>:" \
  -H "Stripe-Account: <merchant account ID>" \
     # Example: my.domain.com
  -d domain_name="<your_url>"
```

3. Enable ECE buttons by setting the `_wcpay_wcpay_feature_stripe_ece` to `1`: `wp option set _wcpay_feature_stripe_ece 1`.
4. Keep a tab open on **wp-admin &rarr; Payments &rarr; Settings &rarr; Express checkouts &rarr; Apple Pay/Google Pay &rarr; Customize**. You'll be making changes there during the tests.
5. Add a simple product to your cart.

### Test button sizes

> [!NOTE]
> Due to limitations in Stripe's API the maximum height of the buttons is `55px` instead of the `56px` suggested by the `Large` option in the settings. We're discussing this in https://github.com/Automattic/woocommerce-payments/issues/8951#issuecomment-2183177946.

1. Select a size in the settings.
2. Open the shortcode Cart page.
3. Inspect the button using your browser's developer tools.
4. Make sure the height matches the setting.
5. Repeat steps (2)-(4) for the shortcode Checkout, and Blocks Cart and Checkout pages.
6. Repeat steps (1)-(5) for the other 2 sizes.

### Test button theme (i.e. color)

> [!IMPORTANT]
> Stripe only supports an `Outline` version of the Google Pay button so both the `Light` and `Outline` themes will have an outline.

> [!NOTE]
> You can keep the size the same while doing this test.

1. Select a theme in the settings.
2. Open the shortcode Cart page.
3. Make sure the button theme matches the setting.
5. Repeat steps (2)-(4) for the shortcode Checkout, and Blocks Cart and Checkout pages.
6. Repeat steps (1)-(5) for the other 2 sizes.

### Test combinations

- Test other combinations of themes and sizes (not necessarily exhaustive, pick 2-3 additional combinations you haven't tested) and make sure the settings are applied correctly.

-------------------


**Post merge**



- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. : _Not Applicable_
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. _Not Applicable_
